Tooltip V0.2 with portals

dancingshell opened the pull request:
### Summary
<!--Why you made the changes-->
<!--What is the intended result-->
* new API with portals!
* tooltips can now be controlled (not hoverable, show prop passed in by parent) allowing it to stay open

### Screenshots
![awesomescreenshot-2017-11-17t19-11-32-172z](https://user-images.githubusercontent.com/6993945/32964514-58c13708-cb88-11e7-9c17-71f0bf54fa53.gif)

ben-burlingham commented on tooltip/src/Tooltip.jsx:
```diff
@@ -1,130 +1,126 @@
+
+  static propTypes = {
+    controlled: PropTypes.bool,
+    placement: PropTypes.oneOf(['top', 'bottom', 'left', 'right']).isRequired,
```
nb Could these be expressed as an enum, and exported?

mcclayton commented on tooltip/src/Tooltip.jsx:
```diff
@@ -1,130 +1,126 @@
+    controlled: PropTypes.bool,
+    placement: PropTypes.oneOf(['top', 'bottom', 'left', 'right']).isRequired,
+    show: PropTypes.bool,
+    title: PropTypes.string.isRequired
```
`nb` I'd consider generalizing this to something like `content`. This gives us the flexibility to turn this into `Proptypes.node` in the future without needing a breaking change and makes it a little more general.

mcclayton commented on tooltip/src/Tooltip.jsx:
```diff
@@ -1,130 +1,126 @@
+import TooltipPortal from './TooltipPortal';
+import styles from './Tooltip.scss';
 
+const ADJUSTMENT = 8;
```
Is this the padding between the tooltip pointer and the component it's wrapping?
Might be nice to add a comment if so.

dancingshell replied:
Removed and re-implemented with css

mcclayton commented on tooltip/src/Tooltip.jsx:
```diff
@@ -1,130 +1,126 @@
-    this.setState({ title: '' });
-  }
+  onMouseEnter = () => {
+    if (!this.props.controlled) {
```
I love the idea of being able to control the tooltip from a parent or have the state self-managed.
In the future I could easily see the self-managed tooltip state being managed by a HoC. This seems to be equally great though :)

mcclayton commented on tooltip/src/Tooltip.jsx:
```diff
@@ -1,130 +1,126 @@
+
+  position() {
+    if (this.parentElement && this.tooltip) {
+      const {
```
`nb`: Not blocking - just pointing out that not all of these properties are available to older browsers:
Here's an example of what IE 8 vs. 9 returns -> https://stackoverflow.com/a/17686282
Not blocking though.

mcclayton commented:
You'll likely want a throttled window resize handler to help reposition the tooltip on window resize. This will be evident for controlled tooltips.

mcclayton commented:
Final thing I'd call out (all of my comments are non-blocking)
is that the styling is slightly different from what @nickwmurphy had.
<img width="256" alt="screen shot 2017-11-17 at 11 55 20 am" src="https://user-images.githubusercontent.com/2695054/32966379-80f42680-cb8e-11e7-8084-05d95fab0ab1.png">
<img width="326" alt="screen shot 2017-11-17 at 11 56 00 am" src="https://user-images.githubusercontent.com/2695054/32966380-810ddf80-cb8e-11e7-8799-0e62f3b8eb68.png">

This may be intentional, but calling out here in case it wasn't.

dancingshell commented:
@mcclayton @nickwmurphy @ben-burlingham 
Updates
- Added `popover` prop that applies popover styling
- Export position options
- Controlled tooltips now reposition on page resize
- Implemented old adjusted position var with css

mcclayton commented on tooltip/src/Tooltip.jsx:
```diff
@@ -63,7 +85,17 @@ class TooltipWrapper extends React.Component {
+  }
+
   position() {
+    console.log('trying')
```
^ console

dancingshell replied:
dang 🙃

dancingshell commented:
also here is the popover styles 
![popover](https://user-images.githubusercontent.com/6993945/32972045-0e3ee83e-cba5-11e7-99e9-0e721a0996f7.png)
